lib/src/http_client.dart

```diff
@@ -0,0 +1,281 @@
+  ///
+  /// Returns a [Future] of [HttpClientRequest].
+  Future<HttpClientRequest> getUrl(Uri url) {
+    return openUrl('GET', url);
+  }
+
+  /// Opens a request on the basis of [host], [port] and [path] using GET method.
+  ///
+  /// Returns a [Future] of [HttpClientRequest].
+  Future<HttpClientRequest> get(String host, int port, String path) {
+    return getUrl(_getUri(host, port, path));
+  }
+
+  /// Opens a [url] using HEAD method.
+  ///
+  /// Returns a [Future] of [HttpClientRequest].
+  Future<HttpClientRequest> headUrl(Uri url) {
+    return openUrl('HEAD', url);
+  }
+
+  /// Opens a request on the basis of [host], [port] and [path] using HEAD method.
+  ///
+  /// Returns a [Future] of [HttpClientRequest].
+  Future<HttpClientRequest> head(String host, int port, String path) {
+    return headUrl(_getUri(host, port, path));
+  }
+
+  /// Opens a [url] using PUT method.
+  ///
+  /// Returns a [Future] of [HttpClientRequest].
+  Future<HttpClientRequest> putUrl(Uri url) {
+    return openUrl('PUT', url);
+  }
+
+  /// Opens a request on the basis of [host], [port] and [path] using PUT method.
+  ///
+  /// Returns a [Future] of [HttpClientRequest].
+  Future<HttpClientRequest> put(String host, int port, String path) {
+    return putUrl(_getUri(host, port, path));
+  }
+
+  /// Opens a [url] using POST method.
+  ///
+  /// Returns a [Future] of [HttpClientRequest].
+  Future<HttpClientRequest> postUrl(Uri url) {
+    return openUrl('POST', url);
+  }
+
+  /// Opens a request on the basis of [host], [port] and [path] using POST method.
+  ///
+  /// Returns a [Future] of [HttpClientRequest].
+  Future<HttpClientRequest> post(String host, int port, String path) {
+    return postUrl(_getUri(host, port, path));
+  }
+
+  /// Opens a [url] using PATCH method.
+  ///
+  /// Returns a [Future] of [HttpClientRequest].
+  Future<HttpClientRequest> patchUrl(Uri url) {
+    return openUrl('PATCH', url);
+  }
+
+  /// Opens a request on the basis of [host], [port] and [path] using PATCH method.
+  ///
+  /// Returns a [Future] of [HttpClientRequest].
+  Future<HttpClientRequest> patch(String host, int port, String path) {
+    return patchUrl(_getUri(host, port, path));
+  }
+
+  /// Opens a [url] using DELETE method.
+  ///
+  /// Returns a [Future] of [HttpClientRequest].
+  Future<HttpClientRequest> deleteUrl(Uri url) {
+    return openUrl('DELETE', url);
+  }
+
+  /// Opens a request on the basis of [host], [port] and [path] using DELETE method.
+  ///
+  /// Returns a [Future] of [HttpClientRequest].
+  Future<HttpClientRequest> delete(String host, int port, String path) {
+    return deleteUrl(_getUri(host, port, path));
+  }
+
+  /// Function for resolving the proxy server to be used for a HTTP connection from
+  /// the proxy configuration specified through environment variables.
+  ///
+  /// Note: It just returns `dart:io` [HttpClient.findProxyFromEnvironment].
+  static String findProxyFromEnvironment(Uri url,
```

Comment:
This is unused in our code base. Should we remove it?

If it serves a purpose, add a test that uses this.

Reply:
Unused. Just here for compatibility reasons. Should I remove it?

Reply:
Yeah, just remove it for now.

We can always add it back in when people start using `package:cronet` and find it missing. (But removing things later from an API is always problematic.)